include/modem/at_monitor.rst

```diff
@@ -0,0 +1,115 @@
+
+Initialization
+==============
+
+The application can initialize the AT monitor library manually via the :c:func:`at_monitor_init` function or automatically, via the ``SYS_INIT`` macro, by enabling :option:`CONFIG_AT_MONITOR_SYS_INIT`.
+Upon initialization, the AT monitor library will register itself as the reicever of AT notifications from the modem library (via :c:func:`nrf_modem_at_notif_handler_set`).
+
+.. note::
+   When the AT monitor library is initialized, it will override whatever handler had been previously set with the modem library to receive the AT notifications.
+
+Usage
+=====
+
+The application can define an AT monitor to receive notifications via the :c:macro:`AT_MONITOR` macro.
+An AT monitor has a name, a filter string, and a callback function. In addition, it may be paused or active (:c:macro:`PAUSED`, :c:macro:`ACTIVE` respecively).
+When the AT monitor library receives an AT notification from the modem library, the notification is copied on the AT monitor library heap and is dispatched via the system workqueue to all monitors whose filter matches (even partially) the contents of the notification.
+Multiple parts of the application can define their own AT monitor with the same filter as another AT monitor, and thus receive the same notifications, if desired.
+
+The following code snippet shows how to register a handler that receives ``+CEREG`` notifications from the modem library.
+
+.. code-block:: c
+
+	/* AT monitor for +CEREG notifications */
+	AT_MONITOR(network_registration, "+CEREG", cereg_mon);
+
+	int cereg_mon(const char *notif)
+	{
+		printf("Received +CEREG notification: %s", notif);
+	}
+
+Pausing and resuming
+********************
+
+When defined, an AT monitor is in active state by default. An AT monitor may be paused and resumed with the :c:func:`at_monitor_pause` and :c:func:`at_monitor_resume` functions respectively.
+If desired, an AT monitor can be defined to be in paused state at compile time by appending :c:macro:`PAUSED` to the monitor definition.
+
+The following code snippet shows how to define an AT monitor for ``+CEREG`` notifications that is paused at boot, and later resumed.
+
+.. code-block:: c
+
+	/* AT monitor for +CEREG notifications, paused until manually activated */
+	AT_MONITOR(network_registration, "+CEREG", cereg_mon, PAUSED);
+
+	void foo(void)
+	{
+		/* let's resume the monitor */
+		at_monitor_resume(network_registration);
+	}
+
+Wildcard filter
+***************
+
+It is possible to define an AT monitor that will receive all AT notifications, by passing :c:macro:`ANY` as the AT monitor filter string.
+
+The following code snippet shows how to define an AT monitor that will receive all AT notifications.
+
+.. code-block:: c
+
+	/* AT monitor for all notifications */
+	AT_MONITOR(catch_all, ANY, at_notif_handler);
+
+	int at_notif_handler(const char *notif)
+	{
+		printf("Received a notification: %s", notif);
+	}
+
+Relation to the AT notification library
```

Comment:
Do we have some preference on whether users should use at_notif library or at_monitor library? If yes, I'd write it down here, and maybe also to at_notif documentation.

Reply:
I added a line at the end of this section. It is recommended to use `at_monitor` for new applications since, `at_notif` is being deprecated. Existing applications will have to port to `at_monitor` at some point.